unit_tests/testVTKFiles/testVTKFiles.f90

```diff
@@ -303,32 +351,112 @@ PROGRAM testVTKFiles
          ALL((testVTKMesh%y .APPROXEQ. yref+2.0_SRK)) .AND. &
          ALL((testVTKMesh%z .APPROXEQ. zref+3.0_SRK))
   ASSERT(bool, 'testVTKMesh%translate(...) UNSTRUCTURED_GRID')
+
+ENDSUBROUTINE testMeshTranslation
 !
-!Test mesh addition
+!-------------------------------------------------------------------------------
+SUBROUTINE testMeshAddition()
+  INTEGER(SIK) :: i
+  TYPE(VTKMeshType),ALLOCATABLE :: mesh_array(:)
+
+  COMPONENT_TEST('Binary')
   CALL testVTKFile%clear()
   DEALLOCATE(xref)
   DEALLOCATE(yref)
   DEALLOCATE(zref)
   CALL SetupTest14_Mesh()
   testVTKMesh=testVTKMesh+testVTKMesh2
-  CALL testVTKMesh%cleanupPoints()
   CALL testVTKFile%initialize(666,'testVTK14.vtk',STATUS='testVTK14')
   CALL testVTKFile%writeMesh(testVTKMesh)
-  ALLOCATE(xref(12))
-  ALLOCATE(yref(12))
-  ALLOCATE(zref(12))
-  xref=(/ 0.0,0.0,0.0,0.0,0.0,0.5,0.5,1.0,1.0,1.0,1.0,1.0 /)
-  yref=(/ 0.0,0.0,0.5,1.0,1.0,0.5,0.5,0.0,0.0,0.5,1.0,1.0 /)
-  zref=(/ 0.0,1.0,1.5,0.0,1.0,0.0,1.0,0.0,1.0,1.5,0.0,1.0 /)
-  bool = ALL((testVTKMesh%x .APPROXEQ. xref)) .AND. &
-         ALL((testVTKMesh%y .APPROXEQ. yref)) .AND. &
-         ALL((testVTKMesh%z .APPROXEQ. zref))
-  ASSERT(bool, 'VTKMeshType OPERATOR(+)')
+
+  ALLOCATE(xref(30))
+  ALLOCATE(yref(30))
+  ALLOCATE(zref(30))
+  ALLOCATE(cellRef(5))
+  ALLOCATE(nodeRef(30))
+  xref=(/0.0_SRK,1.0_SRK,0.5_SRK,0.0_SRK,1.0_SRK,0.5_SRK,0.0_SRK,0.0_SRK,0.5_SRK,0.0_SRK,0.0_SRK, &
+    0.5_SRK,0.0_SRK,1.0_SRK,0.5_SRK,0.0_SRK,1.0_SRK,0.5_SRK,1.0_SRK,1.0_SRK,0.5_SRK,1.0_SRK,1.0_SRK, &
+    0.5_SRK,0.0_SRK,0.0_SRK,0.0_SRK,1.0_SRK,1.0_SRK,1.0_SRK/)
+  yref=(/0.0_SRK,0.0_SRK,0.5_SRK,0.0_SRK,0.0_SRK,0.5_SRK,0.0_SRK,1.0_SRK,0.5_SRK,0.0_SRK,1.0_SRK, &
+    0.5_SRK,1.0_SRK,1.0_SRK,0.5_SRK,1.0_SRK,1.0_SRK,0.5_SRK,0.0_SRK,1.0_SRK,0.5_SRK,0.0_SRK,1.0_SRK, &
+    0.5_SRK,0.0_SRK,1.0_SRK,0.5_SRK,0.0_SRK,1.0_SRK,0.5_SRK/)
+  zref=(/0.0_SRK,0.0_SRK,0.0_SRK,1.0_SRK,1.0_SRK,1.0_SRK,0.0_SRK,0.0_SRK,0.0_SRK,1.0_SRK,1.0_SRK, &
+    1.0_SRK,0.0_SRK,0.0_SRK,0.0_SRK,1.0_SRK,1.0_SRK,1.0_SRK,0.0_SRK,0.0_SRK,0.0_SRK,1.0_SRK,1.0_SRK, &
+    1.0_SRK,1.0_SRK,1.0_SRK,1.5_SRK,1.0_SRK,1.0_SRK,1.5_SRK/)
+  cellRef=(/13,13,13,13,13/)
+  nodeRef=(/0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29/)
+  ASSERT(testVTKMesh%isInit,'%isInit')
+  ASSERT_EQ(testVTKMesh%meshType,VTK_UNSTRUCTURED_GRID,'%meshType')
+  ASSERT_EQ(testVTKMesh%numPoints,30,'%numPoints')
+  ASSERT_EQ(testVTKMesh%numCells,5,'%numCells')
+  DO i=1,30
+    ASSERT_APPROXEQ(testVTKMesh%x(i),xref(i),'%x')
+    ASSERT_APPROXEQ(testVTKMesh%y(i),yref(i),'%y')
+    ASSERT_APPROXEQ(testVTKMesh%z(i),zref(i),'%z')
+    ASSERT_EQ(testVTKMesh%nodeList(i),nodeRef(i),'%nodeList')
+  ENDDO !i
+  DO i=1,5
+    ASSERT_EQ(testVTKMesh%cellList(i),cellRef(i),'%cellList')
+  ENDDO !i
+
   DEALLOCATE(xref)
   DEALLOCATE(yref)
   DEALLOCATE(zref)
+  DEALLOCATE(cellRef)
+  DEALLOCATE(nodeRef)
+
+  COMPONENT_TEST('Array')
```

Comment:
This COMPONENT_TEST is for the new routine I added.  Everything else was just rearranging to use the test harness better and update to follow the new code standard.